src/boilerplate.rs

```diff
@@ -1,22 +1,87 @@
 use std::fs;
+use std::path::Path;
 
 use crate::html;
 use crate::spec::Spec;
 
+// Retrieve boilerplate file with doc (metadata).
+fn retrieve_boilerplate(doc: &mut Spec, name: &str) -> String {
+    retrieve_boilerplate_with_info(name, doc.md.group.clone(), doc.md.raw_status.clone())
+}
+
+// Retrieve boilerplate file with group and status.
+fn retrieve_boilerplate_with_info(
+    name: &str,
+    group: Option<String>,
+    status: Option<String>,
+) -> String {
+    // File Priorities:
+    // 1. [status file with group]
+    // 2. [generic file with group]
+    // 3. [status file without group]
+    // 4. [generic file without group]
+
+    let mut paths_to_try = Vec::new();
+
+    let status_filename = if let Some(status) = status {
+        Some(format!("{}-{}.include", name, status))
+    } else {
+        None
+    };
+
+    if let Some(ref status_filename) = status_filename {
+        // status file with group
+        if let Some(group) = group.clone() {
+            paths_to_try.push(Path::new("boilerplate").join(group).join(status_filename));
+        }
+    }
+
+    let generic_filename = format!("{}.include", name);
+
+    if let Some(group) = group {
+        // generic file with group
+        paths_to_try.push(
+            Path::new("boilerplate")
+                .join(group)
+                .join(generic_filename.clone()),
+        );
+    }
+
+    if let Some(ref status_filename) = status_filename {
+        // status file without group
+        paths_to_try.push(Path::new("boilerplate").join(status_filename));
+    }
+
+    // generic file without group
+    paths_to_try.push(Path::new("boilerplate").join(generic_filename));
+
+    for path in paths_to_try {
+        if path.is_file() {
+            match fs::read_to_string(path) {
+                Ok(data) => {
+                    return data;
+                }
+                Err(_) => die!("Can't read data from the include file for {}.", name),
+            }
+        }
+    }
+
+    die!("Can't find an appropriate include file for {}.", name);
+}
+
 pub fn add_header_footer(doc: &mut Spec) {
-    // TODO: handle group and status
-    let header_path = if doc.md.boilerplate.get("header") {
-        "boilerplate/header.include"
+    let header = if doc.md.boilerplate.get("header") {
+        retrieve_boilerplate(doc, "header")
     } else {
-        ""
+        String::new()
     };
-    let footer_path = if doc.md.boilerplate.get("footer") {
-        "boilerplate/footer.include"
+
+    let footer = if doc.md.boilerplate.get("footer") {
```

Comment:
might be better to move the `boilerplate.get()` call inside `retrieve_boilerplate`?

So:

```
if !doc.md.boilerplate.get(name) {
    return String::new();
}
// ...
```

Then these become simpler.